### PR TITLE
PWGDQ: Add diElectronsAll table

### PR DESCRIPTION
**Pull request by lupengzhong**
Add the diElectronsAll table.
There is the configurable "fConfigFlatTables" to choose whether to output the diElectronsAll table.

### CODE REVIEW

File: PWGDQ/Core/VarManager.h [hunk 1]
```diff
@@ -1024,8 +1034,8 @@ KFPVertex VarManager::createKFPVertexFromCollision(const T& collision)
   kfpVertex.SetXYZ(collision.posX(), collision.posY(), collision.posZ());
   kfpVertex.SetCovarianceMatrix(collision.covXX(), collision.covXY(), collision.covYY(), collision.covXZ(), collision.covYZ(), collision.covZZ());
   kfpVertex.SetChi2(collision.chi2());
-  kfpVertex.SetNDF(2 * collision.multNTracksPV() - 3);
-  kfpVertex.SetNContributors(collision.multNTracksPV());
+  kfpVertex.SetNDF(2 * collision.numContrib() - 3);
+  kfpVertex.SetNContributors(collision.numContrib());
```
Comment (iarsene) [inline PWGDQ/Core/VarManager.h:1038]: why is this change needed ?

File: PWGDQ/Core/VarManager.h [hunk 2]
```diff
@@ -2566,13 +2576,13 @@ void VarManager::FillPairVertexing(C const& collision, T const& t1, T const& t2,
...
       values[kVertexingTauz] = (collision.posZ() - secondaryVertex[2]) * v12.M() / (TMath::Abs(v12.Pz()) * o2::constants::physics::LightSpeedCm2NS);
-      values[kVertexingTauxy] = values[kVertexingLxy] * v12.M() / (v12.P() * o2::constants::physics::LightSpeedCm2NS);
+      values[kVertexingTauxy] = values[kVertexingLxy] * v12.M() / (TMath::Sqrt((v12.Px() * v12.Px()) + (v12.Py() * v12.Py())) * o2::constants::physics::LightSpeedCm2NS);
 
       values[kVertexingPz] = TMath::Abs(v12.Pz());
       values[kVertexingSV] = secondaryVertex[2];
 
       values[kVertexingTauzErr] = values[kVertexingLzErr] * v12.M() / (TMath::Abs(v12.Pz()) * o2::constants::physics::LightSpeedCm2NS);
-      values[kVertexingTauxyErr] = values[kVertexingLxyErr] * v12.M() / (v12.P() * o2::constants::physics::LightSpeedCm2NS);
+      values[kVertexingTauxyErr] = values[kVertexingLxyErr] * v12.M() / (TMath::Sqrt((v12.Px() * v12.Px()) + (v12.Py() * v12.Py())) * o2::constants::physics::LightSpeedCm2NS);
```
Comment (iarsene) [inline PWGDQ/Core/VarManager.h:2585]: here you should use the pt from the computation I suggested above

File: PWGDQ/Core/VarManager.h [hunk 3]
```diff
@@ -2924,13 +2966,13 @@ void VarManager::FillDileptonTrackVertexing(C const& collision, T1 const& lepton
       }
 
       values[kVertexingTauz] = (collision.posZ() - secondaryVertex[2]) * v123.M() / (TMath::Abs(v123.Pz()) * o2::constants::physics::LightSpeedCm2NS);
-      values[kVertexingTauxy] = values[kVertexingLxy] * v123.M() / (v123.P() * o2::constants::physics::LightSpeedCm2NS);
+      values[kVertexingTauxy] = values[kVertexingLxy] * v123.M() / (TMath::Sqrt((v123.Px() * v123.Px()) + (v123.Py() * v123.Py())) * o2::constants::physics::LightSpeedCm2NS);
```
Comment (iarsene) [inline PWGDQ/Core/VarManager.h:2969]: same comment as above

File: PWGDQ/Core/VarManager.h [hunk 4]
```diff
@@ -2924,13 +2966,13 @@ void VarManager::FillDileptonTrackVertexing(C const& collision, T1 const& lepton
...
 
       values[kVertexingPz] = TMath::Abs(v123.Pz());
       values[kVertexingSV] = secondaryVertex[2];
 
       values[kVertexingTauzErr] = values[kVertexingLzErr] * v123.M() / (TMath::Abs(v123.Pz()) * o2::constants::physics::LightSpeedCm2NS);
-      values[kVertexingTauxyErr] = values[kVertexingLxyErr] * v123.M() / (v123.P() * o2::constants::physics::LightSpeedCm2NS);
+      values[kVertexingTauxyErr] = values[kVertexingLxyErr] * v123.M() / (TMath::Sqrt((v123.Px() * v123.Px()) + (v123.Py() * v123.Py())) * o2::constants::physics::LightSpeedCm2NS);
```
Comment (iarsene) [inline PWGDQ/Core/VarManager.h:2975]: same comment as above

File: PWGDQ/Core/VarManager.h (hunk 1 above)
Comment (lupengzhong) [reply PWGDQ/Core/VarManager.h:1038]: Hi Ionut,
This is a bug fix when creating the KF object for the PV. In principle, the variable need to be used to set the NDF and NContributors to the PV is how many tracks were used to reconstruct the PV. By mistake, the `multNTracksPV` which means removing the number of the secondary tracks from the `numContrib` was used.

File: PWGDQ/Core/VarManager.h [hunk 5]
```diff
@@ -2566,13 +2576,13 @@ void VarManager::FillPairVertexing(C const& collision, T const& t1, T const& t2,
       values[kVertexingLxyz] = std::sqrt(values[kVertexingLxyz]);
 
       values[kVertexingTauz] = (collision.posZ() - secondaryVertex[2]) * v12.M() / (TMath::Abs(v12.Pz()) * o2::constants::physics::LightSpeedCm2NS);
-      values[kVertexingTauxy] = values[kVertexingLxy] * v12.M() / (v12.P() * o2::constants::physics::LightSpeedCm2NS);
+      values[kVertexingTauxy] = values[kVertexingLxy] * v12.M() / (TMath::Sqrt((v12.Px() * v12.Px()) + (v12.Py() * v12.Py())) * o2::constants::physics::LightSpeedCm2NS);
```
[the remark this answers is not in the document]
Comment (lupengzhong) [reply PWGDQ/Core/VarManager.h]: Get the pT with the v12.Pt() function.

File: PWGDQ/Core/VarManager.h (hunk 2 above)
Comment (lupengzhong) [reply PWGDQ/Core/VarManager.h]: Same as above

File: PWGDQ/Core/VarManager.h (hunk 3 above)
Comment (lupengzhong) [reply PWGDQ/Core/VarManager.h]: Same as above

File: PWGDQ/Core/VarManager.h (hunk 4 above)
Comment (lupengzhong) [reply PWGDQ/Core/VarManager.h]: Same as above

**Review by iarsene — APPROVED** [review]
